pkg/virt-controller/watch/vmi.go

```diff
@@ -482,6 +482,25 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 					break
 				}
 
+				if controller.VMIHasHotplugVolumes(vmi) {
+					hotplugAttachmentPods, err := controller.AttachmentPods(pod, c.podInformer)
+					if err != nil {
+						syncErr = &syncErrorImpl{fmt.Errorf("failed to get attachment pods: %v", err), FailedHotplugSyncReason}
+						break
+					}
+					attachmentPodReady := false
+					for _, attachmentPod := range hotplugAttachmentPods {
+						if isPodReady(attachmentPod) && attachmentPod.DeletionTimestamp == nil && attachmentPod.Spec.NodeName == pod.Spec.NodeName {
+							attachmentPodReady = true
+							break
+						}
+					}
+					if !attachmentPodReady {
+						log.Log.V(3).Object(vmi).Infof("Postpone the pod hand-over and await the attachment pod ready")
+						break
+					}
+				}
+
```

Comment:
can this be abstracted into a helper function? something like `hotplugPodsReady(vmi)` which is a no-op if no hotplug volumes exist, or finds the attachment pod and makes sure it's ready if hot plug volumes do exist. 

i'm just trying to get us into the habit of reducing the size of some of our huge functions. We've been eating away at improving the situation a bit in refactors as well.

Reply:
Fixed here 🙂
https://github.com/kubevirt/kubevirt/pull/6079/commits/f20617ff5b3de6a3a6f92c76cdad431735195fb2#diff-16487fff595728eb5eeec01225a2203756ecad25d90e90d5feb13a9ff08df7ffR812-R827